src/physics/euler.cpp

```diff
@@ -390,16 +384,18 @@ template<typename real2>
 inline real2 Euler<dim,nstate,real>
 :: compute_entropy (const real2 density, const real2 pressure) const
 {
-    real2 entropy = pressure * pow(density, -gam);
-    if (entropy > 0){
-        entropy = log( entropy );
+    // copy density and pressure such that the check will not modify originals
+    real2 density_check = density; 
+    real2 pressure_check = pressure;
+    const bool density_is_positive = check_positive_quantity(density_check, "density");
+    const bool pressure_is_positive = check_positive_quantity(pressure_check, "pressure");
+    if (density_is_positive && pressure_is_positive) {
+        real2 entropy = pressure * pow(density, -gam);
```

Comment:
call `compute_entropy_measure` ?

Reply:
I would rather leave that explicitly written. To me it's confusing to refer to "entropy measure" for this simple equation.